### PR TITLE
#3530 Password requirements should be displayed when password requirements is enabled

### PR DESCRIPTION
**Pull request by vestrada**
https://github.com/craftercms/craftercms/issues/3530 - [studio-ui] Password requirements should be displayed when password requirements is enabled

### CODE REVIEW

File: static-assets/scripts/main.js [hunk 1]
```diff
@@ -716,9 +720,113 @@
...
+            } else {
+              html += '<span class="list-icon fa fa-times-circle red "></span>';
+              validPass = true;
+            }
+          }
+          html += messages[captureGroupName[0].replace(/\?<|>/g, "")] +
+            '</li>';
+        });
+        html += '</ul>';
+        if (staticTemplate) {
```
Comment (rart) [inline static-assets/scripts/main.js:782]: single if with dual. condittion

File: static-assets/scripts/main.js [hunk 2]
```diff
@@ -716,9 +720,113 @@
...
+            html += '<p class="result red"><span class="list-icon fa fa-times-circle"></span>' + messages.invalidPassword + '</p>';
+            validPass = true;
+          }
+        }
+        html += '</div>';
+        return { 'template': html, 'validPass': validPass };
+      }
+
+      this.runValidation = function ($scope, elt, staticTemplate) {
+        $(function () {
```
Comment (rart) [inline static-assets/scripts/main.js]: no need for the doc ready wrap

File: ui/app/src/utils/i18n-legacy.ts [hunk 3]
```diff
@@ -88,7 +88,50 @@ const usersAdminMessages = defineMessages({
...
+  validPassword: {
+    id: 'passwordRequirement.validPassword',
+    defaultMessage: 'Requirements met'
+  },
+  invalidPassword: {
+    id: 'passwordRequirement.invalidPassword',
+    defaultMessage: 'Requirements not met'
+  },
+  fullfillAllReqErrorMessage: {
+    id: 'passwordRequirement.fullfillAllReqErrorMessage',
```
Comment (rart) [inline ui/app/src/utils/i18n-legacy.ts]: fulfill

File: ui/scss/user-dashboard/_admin.scss [hunk 4]
```diff
@@ -242,6 +242,33 @@ table {
   }
 }
 
+.password-popover {
+  ul{
```
Comment (rart) [inline ui/scss/user-dashboard/_admin.scss]: use classes unnested classes (à la BEM)

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/utils/i18n-legacy.ts (hunk 3 above)
Comment (vestrada) [reply ui/app/src/utils/i18n-legacy.ts]: Done

File: ui/scss/user-dashboard/_admin.scss (hunk 4 above)
Comment (vestrada) [reply ui/scss/user-dashboard/_admin.scss]: Done

File: static-assets/scripts/main.js (hunk 1 above)
Comment (vestrada) [reply static-assets/scripts/main.js:782]: Else is only for the second condition. It doesn't depend on first if (condition)

File: static-assets/scripts/main.js (hunk 2 above)
Comment (vestrada) [reply static-assets/scripts/main.js]: Done

**Review by rart — CHANGES_REQUESTED** [review]
Password confirm field shadow should also be changed to red together with the border whilst confirm string doesn't match main password.

**Review by rart — APPROVED** [review]

**Pull request merged**